rtv/content.py

```diff
@@ -367,52 +385,96 @@ def __init__(self, name, submissions, loader, order=None):
             raise exceptions.SubredditError('No submissions')
 
     @classmethod
-    def from_name(cls, reddit, name, loader, order=None, query=None):
+    def from_name(cls, reddit, name, loader, order=None, query=None,
+                  listing='r', period=None):
+
+        # Strip leading, trailing, and redundant backslashes
+        n = ''
+        n = ''.join([n + ''.join(list(g)) if k != '/' else '/' \
+                                       for k, g in groupby(name)]).strip(' /')
```

Comment:
Oh I see. I've never seen anybody use ``groupby`` like that before!

Here's an alternative method that may be clearer
```
name = '/'.join(seg for seg in name.split('/) if seg)
```